### PR TITLE
Initial attempt at contribution, submission docs, and templates

### PR DESCRIPTION
**Pull request by newky2k**
Initial attempt at contribution, submission docs, and templates files.

Copied from Xamarin.Forms and reworked to Caboodle.

Starting point.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Check out [A Beginner's Guide for Contributing to Microsoft Caboodle](https://blog.xamarin.com/beginners-guide-contributing-xamarin-forms/).
+
+### Bug Fixes
+
+If you're looking for something to fix, please browse [open issues](https://github.com/xamarin/Caboodle/issues). 
+
+Follow the style used by the [.NET Foundation](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md), with two primary exceptions:
+
+- We do not use the `private` keyword as it is the default accessibility level in C#.
+- We use hard tabs over spaces. You can change this setting in VS 2015 via `Tools > Options` and navigating to `Text Editor > C#` and selecting the "Keep tabs" radio option.
```
Comment (Redth) [inline CONTRIBUTING.md]: What's everyone's opinion here? I like spaces over tabs personally.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+Please see our [Code of Conduct](CODE_OF_CONDUCT.md).
+
+## Prerequisite
+
+You will need to complete a Contribution License Agreement before any pull request can be accepted. Complete the CLA at https://cla2.dotnetfoundation.org/.
+
+## Contributing Code
+
+Check out [A Beginner's Guide for Contributing to Microsoft Caboodle](https://blog.xamarin.com/beginners-guide-contributing-xamarin-forms/).
```
Comment (mattleibow) [inline CONTRIBUTING.md]: This is a Forms link, we may have to create a wiki or something.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+### Bug Fixes
+
+If you're looking for something to fix, please browse [open issues](https://github.com/xamarin/Caboodle/issues). 
+
+Follow the style used by the [.NET Foundation](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md), with two primary exceptions:
```
Comment (Redth) [inline CONTRIBUTING.md]: I also think we should not use `_variable` or `s_variable` as is their convention.  Thoughts?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+###### What "Accepted" does not mean
+* The proposal will ever be implemented, either by a community member or by the core Microsoft Caboodle team.
+* The core Microsoft Caboodle team is committing to implementing a proposal, even if nobody else does. Accepted proposals simply mean that the core Microsoft Caboodle team and the community agree that this proposal should be a part of Microsoft Caboodle.
+
+##### In Progress
+Once a developer has begun work on a proposal, either from the core Microsoft Caboodle team or a community member, the proposal is marked as in progress with the implementors name and (possibly) a link to a development branch to follow along with progress.
+
+#### Rejected
+Rejected proposals will not be implemented or merged into Microsoft Caboodle. Once a proposal is rejected, the thread will be closed and the conversation is considered completed, pending considerable new information or changes..
```
Comment (mattleibow) [inline CONTRIBUTING.md]: double period at the end of the sentence

File: PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+List all API changes here (or just put None), example:
+
+Added: 
+ 
+- string - string Class.Property { get; set; } 
+- void Class.Method();
+
+Changed:
+
+ - object Cell.OldPropetyName => object Cell.NewPropetyName
```
Comment (mattleibow) [inline PULL_REQUEST_TEMPLATE.md]: "Property" not "Propety"

File: SUBMITTING-ISSUES.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+* In the case of a regression, please confirm as such as well as the last working, pre-regression build you may be aware of, if possible. Providing a reproduction with the working package installed to begin with also helps, as we can then simply upgrade it and compare against the latest stable to confirm.
+
+* In some cases, issues may only occur on specific devices, with specific versions of the OS, or when targeting specific OS versions within the project itself. Some issues also occur on devices, but not on simulators. This information is incredibly helpful.
+
+* Please upload reproductions directly to the report as an attachment; this guarantees future access to the file.
+
+#Android Specific Concerns#
+Please let us know any issues which may stem from different versions of the support library packages (e.g. Xamarin.Android.Support.v7.AppCompat), as sometimes bugs may only occur when specific versions are used.
+iOS Specific Concerns
```
Comment (mattleibow) [inline SUBMITTING-ISSUES.md]: this should be a heading as well

File: CONTRIBUTING.md (hunk 1 above)
Comment (mattleibow) [reply CONTRIBUTING.md]: I used to be a spaces person, but tabs are nice as you can adjust the indenting size based on your own preferences.

Comment (newky2k) [reply CONTRIBUTING.md]: i think the default for VS and VS4M is tabs isn't it?

Comment (jamesmontemagno) [reply CONTRIBUTING.md]: This should say VS 2017 as that is the only ide we support on windows

File: CONTRIBUTING.md (hunk 3 above)
Comment (jamesmontemagno) [reply CONTRIBUTING.md]: Agree. no _

File: CONTRIBUTING.md (hunk 1 above)
Comment (jamesmontemagno) [reply CONTRIBUTING.md]: I believe the default is spaces and we should go with that.

File: SUBMITTING-ISSUES.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+#Android Specific Concerns#
+Please let us know any issues which may stem from different versions of the support library packages (e.g. Xamarin.Android.Support.v7.AppCompat), as sometimes bugs may only occur when specific versions are used.
+
+#iOS Specific Concerns#
+To be added
+
+#UWP-specific Concerns#
+Please make note if an issue appears hardware-specific (slower machines, multiple monitors, etc.), due to the desktop component.
+
+#Contributing#
```
Comment (mattleibow) [inline SUBMITTING-ISSUES.md]: Are these meant to be headings? I think they are missing the spacing to get them to go big and bold.

**Comment by mattleibow** [comment]
build

File: SUBMITTING-ISSUES.md (hunk 7 above)
Comment (newky2k) [reply SUBMITTING-ISSUES.md]: yeah they are, as they are in Xamarin.Forms

Comment (jamesmontemagno) [reply SUBMITTING-ISSUES.md]: Yeah, those should be headers for sure :)

**Pull request merged**